dali/c_api/c_api_test.cc

```diff
@@ -610,6 +636,100 @@ TYPED_TEST(CApiTest, UseCopyKernel) {
   }
 }
 
+
+TYPED_TEST(CApiTest, NoCopyForce) {
+  TensorListShape<> input_shape = {{37, 23, 3}, {12, 22, 3}, {42, 42, 3}, {8, 8, 3},
+                                   {64, 32, 3}, {32, 64, 3}, {20, 20, 3}, {64, 64, 3},
+                                   {10, 10, 3}, {60, 50, 3}, {10, 15, 3}, {48, 48, 3}};
+  TensorList<CPUBackend> input_cpu;
+  input_cpu.Resize(input_shape, TypeInfo::Create<uint8_t>());
+
+  TensorList<TypeParam> input;
+
+  auto device = backend_to_device_type<TypeParam>::value;
+  std::string device_str = device == CPU ? "cpu" : "gpu";
```

Comment:
I would create a function for that.
```suggestion
  std::string device_str = GetDeviceStr(device);
```

Reply:
done